components/Share.vue

```diff
@@ -442,17 +454,25 @@ async function getRecords({query, profileId}) {
     records.push(results[0]);
 
     // remove local credential from type in query and restore its value
-    const index = type.indexOf('OverAgeTokenCredential');
-    type.splice(index, 1);
-    vprQuery.credentialQuery.example.type = type;
+    const index = firstCredentialQueryExampleType.indexOf(
+      'OverAgeTokenCredential'
+    );
+    firstCredentialQueryExampleType.splice(index, 1);
+    firstCredentialQuery.example.type = firstCredentialQueryExampleType;
   }
-  if(vprQuery.credentialQuery.example.type.length === 0) {
+  if(firstCredentialQuery.example.type.length === 0) {
     return records;
   }
-  const {queries: [q]} = await credentialStore.remote.convertVPRQuery({
+  const {queries: q} = await credentialStore.remote.convertVPRQuery({
     vprQuery
   });
-  const {documents: results} = await credentialStore.remote.find({query: q});
+
+  const {documents: results} = await credentialStore.remote.find({
+    // flatten the query: [[{type: A}], [{type: B}, {type: C}], [{type: D}]]
+    //         into   =>  [{type: A}, {type: B}, {type: C}, {type: D}]
+    // so that it can be processed by VerifiableCredentialStore.find()
+    query: q.flat()
+  });
```

Comment:
So I don't think there's a bug with `convertVPRQuery`; it is intended to return back N-many possible queries to be run. This is also evident in the test here: https://github.com/digitalbazaar/bedrock-web-vc-store/blob/v8.0.0/test/web/10-VerifiableCredentialStore.js#L163-L169

I don't think the resulting queries are meant to be flattened into a single query, but rather N-many queries are to be run independently. I think each value in `queries` maps to the same `credentialQuery` in the same order.

Flattening them all together into a single query like the code above does still produces the same result (I think?) but it doesn't allow mapping each particular `credentialQuery` to each result ... as there's just one combined result of everything. If that mapping isn't needed, I guess this approach is ok, but I imagine we probably need that mapping to be able to properly inform the user if their choices will actually match the "and" / "or" conditions of the VPR.